pkg/cmd/miscellaneous.go

```diff
@@ -344,7 +345,36 @@ func isIP(word string) bool {
 }
 
 func getPublicIP() string {
-	result, err := ExecCmdReturnOutput("bash", "-c", "curl ifconfig.me")
+	result, err := ExecCmdReturnOutput("bash", "-c", "curl icanhazip.com")
 	checkError(err)
 	return result
 }
+
+// get role either user or operator
+func getRole() string {
+	var role string
+	// will refactor in https://github.com/gardener/gardenctl/issues/266
+	KUBECONFIG = getKubeConfigOfClusterType("garden")
+	out, err := ExecCmdReturnOutput("bash", "-c", "kubectl --kubeconfig="+KUBECONFIG+" auth can-i get secret -A")
+	if err == nil {
+		if out == "yes" {
+			role = "operator"
+		}
+	} else {
+		role = "user"
+	}
+	return role
+}
+
+func getTechnicalID() string {
```

Comment:
what about the `getShootClusterName` function? It should be removed and this function should be used instead

Reply:
done